src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs

```diff
@@ -755,19 +755,7 @@ private static bool IsArrayCompatibleReadOnlyInterface(Type type)
         private static List<PropertyInfo> GetAllProperties(
             [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)]
             Type type)
-        {
-            var allProperties = new List<PropertyInfo>();
-
-            Type? baseType = type;
-            do
-            {
-                allProperties.AddRange(baseType!.GetProperties(DeclaredOnlyLookup));
-                baseType = baseType.BaseType;
-            }
-            while (baseType != typeof(object));
-
-            return allProperties;
-        }
+        => type.GetProperties(BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance | BindingFlags.Static).ToList();
```

Comment:
(nit) The indention here is non-typical. I would make this one more level of indentation and either move the `=>` up to the previous line or leave it on this line. (Along with collapsing the signature into one line so the method is easier to read.)

```suggestion
            => type.GetProperties(BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance | BindingFlags.Static).ToList();
```